src/mesh/RadioLibInterface.cpp

```diff
@@ -297,8 +298,8 @@ void RadioLibInterface::setTransmitDelay()
     if (p->tx_after) {
         unsigned long add_delay = p->rx_rssi ? getTxDelayMsecWeighted(p->rx_snr) : getTxDelayMsec();
         unsigned long now = millis();
-        p->tx_after = max(p->tx_after + add_delay, now + add_delay);
-        notifyLater(now - p->tx_after, TRANSMIT_DELAY_COMPLETED, false);
+        p->tx_after = min(max(p->tx_after + add_delay, now + add_delay), now + 2 * getTxDelayMsecWeightedWorst(p->rx_snr));
+        notifyLater(p->tx_after - now, TRANSMIT_DELAY_COMPLETED, false);
```

Comment:
Interesting, that `now - p->tx_after` is clearly wrong and could indeed have caused very long delays. 
It shouldn't get higher than `now + 2 * getTxDelayMsecWeightedWorst(p->rx_snr)`, since `tx_after` has been set to `millis() + getTxDelayMsecWeightedWorst()` before, but adding this check can't hurt.

Reply:
> It shouldn't get higher than `now + 2 * getTxDelayMsecWeightedWorst(p->rx_snr)`, since `tx_after` has been set to `millis() + getTxDelayMsecWeightedWorst()` before, but adding this check can't hurt.

It was blowing out, because packet RX storms were interrupting the delay and causing this line to keep adding more delay than the elapsed time since the last time the delay was increased, thus blowing out `tx_after` to ridiculous levels. This change caps the possible blowout to `now + 2 * getTxDelayMsecWeightedWorst(p->rx_snr)` in order to prevent it getting silly.

Reply:
Good point, makes sense.